### PR TITLE
Specify tables with empty key list to be the same as those with no key

### PR DESCRIPTION
**Pull request by jafingerhut**
Intended to resolve this issue https://github.com/p4lang/p4-spec/issues/1120

### CODE REVIEW

**Review by jnfoster — APPROVED** [review]
This reflects the consensus opinion of the August LDWG discussion, so looks good to me!

File: p4-16/spec/P4-16-spec.mdk [hunk 1]
```diff
@@ -6584,9 +6584,10 @@ table is an abstract finite map that is given a key and produces as a
 result either an action or a "miss" indication, as described in
 Section [#sec-mau-semantics].
 
-If a table has no `key` property, then it contains no look-up
-table, just a default action---i.e., the associated lookup table is
-always the empty map.
+If a table has no `key` property, or if the value of its `key`
+property is the empty list, then it contains no look-up table, just a
```
Comment (rst0git) [inline p4-16/spec/P4-16-spec.mdk:6588]: Would it be possible to add an example of "empty list" as in P4 this is not obvious?

Comment (jnfoster) [reply p4-16/spec/P4-16-spec.mdk:6588]: Good idea. Maybe just a parenthetical?

**Review by mbudiu-vmw — APPROVED** [review]

**Review by rst0git — APPROVED** [review]
LGTM

**Comment by mbudiu-vmw** [comment]
We will have to make sure the compiler does this too.

**Pull request merged**